src/core/Akka/Actor/Scheduler/TimerScheduler.cs

```diff
@@ -41,36 +41,39 @@ public Timer(object key, object msg, bool repeat, int generation, ICancelable ta
             }
         }
 
-        public interface ITimerMsg
+        public interface ITimerMsg : IWrappedMessage, INoSerializationVerificationNeeded
         {
             object Key { get; }
             int Generation { get; }
             TimerScheduler Owner { get; }
         }
 
-        private class TimerMsg : ITimerMsg, INoSerializationVerificationNeeded
+        private class TimerMsg : ITimerMsg
         {
             public object Key { get; }
             public int Generation { get; }
             public TimerScheduler Owner { get; }
 
-            public TimerMsg(object key, int generation, TimerScheduler owner)
+            public TimerMsg(object key, int generation, TimerScheduler owner, object message)
             {
-                this.Key = key;
-                this.Generation = generation;
-                this.Owner = owner;
+                Key = key;
+                Generation = generation;
+                Owner = owner;
+                Message = message;
```

Comment:
We're not even going to access this value directly in the main timer use-case, so none of that code path is affected.

All of these messages types are `internal` so there's no public API impact.